game.py

```diff
@@ -242,17 +263,66 @@ def handle_input(self, event):
             elif event.key == pygame.K_RIGHT:
                 self.is_right_pressed = False
 
+            if self.state == PLAY and event.key == pygame.K_SPACE:
+                self.state = PAUSE
+                self.paused_screen()
+
         mos_x, mos_y = pygame.mouse.get_pos()
 
+        pause_rect = self.pause.get_rect(
+            center=((self.middle_of_screen_x - self.background_width * 0.4),
+                    (self.screen_height * 0.065)))
+
+        if (pause_rect.collidepoint(mos_x, mos_y) and
+           event.type == pygame.MOUSEBUTTONUP and
+           event.button == 1 and
+           self.state == PLAY):
+            self.state = PAUSE
+            self.paused_screen()
+
         if (event.type == pygame.MOUSEBUTTONUP and
-           event.button == 1):
+           event.button == 1 and not
+           pause_rect.collidepoint(mos_x, mos_y)):
             if mos_x > self.middle_of_screen_x:
                 self.right_moved = 1
                 self.is_right_pressed = 1
             else:
                 self.left_moved = 1
                 self.is_left_pressed = 1
 
+    def paused_screen(self):
+        # create a semi-transparent black surface
+        overlay_color = pygame.Color(0, 0, 0, 150)
+        overlay_surface = pygame.Surface(
+            (self.screen_width, self.screen_height), pygame.SRCALPHA)
+        overlay_surface.fill(overlay_color)
+
+        self.screen.blit(overlay_surface, (0, 0))
+        pygame.display.update()
+
+        while self.state == PAUSE:
+            while Gtk.events_pending():
+                Gtk.main_iteration()
+            for event in pygame.event.get():
+                if event.type == pygame.QUIT:
+                    pygame.quit()
+                    sys.exit()
```

Comment:
We don't use a `pygame.quit` and `sys.quit` on event type `pygame.QUIT`, you can just use a return.

See the test activity in sugargame.

Reply:
I replaced it with `self.running = False` for code consistency